### PR TITLE
Initiate variables

### PR DESCRIPTION
**Pull request by HillaryM17**
Created variables to be used for functions. "searchInput" is currently hard set to "hello" in order to be able to run API to test, actual element ID is commented out.

### CODE REVIEW

**Review by mattlusty — CHANGES_REQUESTED** [review]
I think you want a Pull into Dev branch (not Main) :P

File: assets/js/script.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+var searchInput = "Hello"; //$("#search");
+var wordDefinitionArea = $(".definition");
```
Comment (mattlusty) [inline assets/js/script.js:2]: Non-critical improvement
Better to identify any unique elements using id attributes (and for consistency).
Although that obviously means changing the HTML (to give these ids).

File: assets/js/script.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+var wordPronunciationArea = $("#audio");
+var wordExamples = $("#examples");
+var historyContainer = $(".previous-searches");
```
Comment (mattlusty) [inline assets/js/script.js:5]: Could we just add a declaration for 

`var favouritesContainer;`

I know there is no HTML for the favouritesContainer yet (and may end up not even implementing it at all), but could we just declare an empty variable for it here anyway. Just so we know what variable name to reference in any "favourites" related functions.

File: assets/js/script.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+var searchInput = "Hello"; //$("#search");
```
Comment (mattlusty) [inline assets/js/script.js:1]: Ah I see why you put this here. I probably helped confuse things... 
As you have partially hinted at already, can we make this the actual element...

`var searchInputElement = $("#search");`

And leave the `var searchInputText` to be defined within the onsearch function (can just hardcode it there for now like you did on this line).

This then means having to complete the full speechURL also within the onsearch() function (see below comment)

File: assets/js/script.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+var wordDefinitionArea = $(".definition");
+var wordPronunciationArea = $("#audio");
+var wordExamples = $("#examples");
+var historyContainer = $(".previous-searches"); 
+var searchButton = $("#search-button"); 
+var history = [];
+var favourites = [];
+
+
+const speechURL = "https://voicerss-text-to-speech.p.rapidapi.com/?key=171ec3cab6f247b4b6e7f596d9171ae7&src=" + searchInput + "&hl=en-us&r=0&c=mp3&f=8khz_8bit_mono";
```
Comment (mattlusty) [inline assets/js/script.js:11]: Following on from above thread...

1) The main issue here, is that we will not have access to the value for searchInput (aka searchInputText) at this point. (That needs adding within the onsearch() function.)

2) Secondly, could we break the speechURL into separate pieces for clarity and easier to update things (ApiKey, and Options etc)...

```
// Speech API Query Parts
const speechBaseURL = "https://voicerss-text-to-speech.p.rapidapi.com/?
const speechApiKey = "key=171ec3cab6f247b4b6e7f596d9171ae7"
const speechOther = "&hl=en-us&r=0&c=mp3&f=8khz_8bit_mono" 
const speechOptions = {
   method: 'GET',
   headers: {
      'X-RapidAPI-Key': '6258e18b25mshd96a594dcf7fcb5p1b1fd6jsn955bae6d8f8b',
      'X-RapidAPI-Host': 'voicerss-text-to-speech.p.rapidapi.com'
   }
};

// Construct Speech API query string 
// Note: the searchInputText is left off the speechURL for now (that should be appended within the onsearch() function)
const speechURL = speechBaseURL + speechApiKey + speechOther; 

// Same needs repeating for the Word API Query 
// Question: how come this word API call does not seem to have / need an apiKey out of curiosity?

```

File: assets/js/script.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+	}
+};
+
+//$(".search-button").on("click", function () {
+    
+  //});
+
+
+  function onsearch() {
+
```
Comment (mattlusty) [inline assets/js/script.js:35]: Following on from same thread above ...

(Technically this is stepping into the onsearch() function feature... but it overlaps with the logic, so hey ho!)

function onsearch () {
   // move the searchInputText assigment here
   let searchInputText = "Hello"

   // Then append the searchInputText to complete the speechURL
   wordURL += "&src=" + searchInputText;

   // Then append the searchInputText to complete the speechURL
   speechURL += "&src=" + searchInputText;

   // then do the fetch()
}

**Comment by HillaryM17** [comment]
If we don't construct the API query call with the SearchInputText I feel it's going to confuse the issue when it comes to creating the whole URL later, splitting it into really small parts is going to cause mistakes to be made surely, especially surrounding order and adding the input,

**Comment by HillaryM17** [comment]
If we don't construct the API query call with the SearchInputText, and split it into seperate variables, I feel it's going to confuse the issue when it comes to creating the whole URL later, especially surrounding the order... 
Additionally, with the WordsURL, I don't know if we can move the SearchInputText input to the end of the URL - it doesn't look like it when I've tried to run it `Endpoint '/words/definitions/Hello' does not exist` 
The wordsURL API also isn't as complicated as the speech one as it only needs the RapidAPI key, not an additional one.  
I don't know if this makes sense written, but happy to have a conversation if it's easier :) I'm free all morning(ish).

**Comment by mattlusty** [comment]
Sorry for delay response. Busy morning. I'll push it through now. 

Your code works, and most the variables are there and ready. So good job. 

Maybe we can discuss a couple of my points when your online.

File: assets/js/script.js [hunk 6]
```diff
@@ -1,20 +1,24 @@
...
-// Hello World!
-
-  
-
-//text to speech API
-const speechURL = 'https://voicerss-text-to-speech.p.rapidapi.com/?key=%3CREQUIRED%3E&src=Hello%2C%20world!&hl=en-us&r=0&c=mp3&f=8khz_8bit_mono';
+var searchInputText = "Hello";
+var searchInputElement = $("#search");
+var wordDefinitionArea = $(".definition");
+var wordPronunciationArea = $("#audio");
```
Comment (mattlusty) [inline assets/js/script.js:4]: To target the audio element (in current HTML structure) ...
`var wordPronunciationArea = $("audio#pronounciation");`
(Translates to audio element with id=pronouncitation)

File: assets/js/script.js [hunk 7]
```diff
@@ -1,20 +1,24 @@
...
+var wordExamples = $("#examples");
+var historyContainer = $(".previous-searches");
```
Comment (mattlusty) [inline assets/js/script.js:6]: The element that contains the previous-search items has id #search-history-items

**Pull request merged**